tests/workflow_contracts/makefile_tooling_test.py

```
@@ -0,0 +1,151 @@
+) -> None:
+    """The AsyncAPI target selects pnpm and passes the validation contract."""
+    env, log_path = fake_tool_environment
+
+    completed = _run_make("lint-asyncapi", env)
+
+    assert completed.returncode == 0, completed.stderr
+    assert _read_invocations(log_path) == [
+        f"pnpm||{REPOSITORY_ROOT / '.uv-cache'}|"
+        f"{REPOSITORY_ROOT / '.uv-tools'}|"
+        "dlx @asyncapi/cli@3.4.2 validate "
+        "spec/asyncapi.yaml --fail-severity=info"
```

Ruff (0.16.0)

[warning] Unparenthesized implicit string concatenation in collection

Did you forget a comma?

(ISC004)